src/main/java/io/jenkins/plugins/sprp/models/Step.java

```diff
@@ -0,0 +1,69 @@
+package io.jenkins.plugins.sprp.models;
+
+import javax.annotation.Nonnull;
+import java.util.AbstractMap;
+import java.util.HashMap;
+
+public class Step {
+    @Nonnull
+    private String stepName;
+    private String defaultParameter;
+    private HashMap<String, Object> parameters;
+
+    Step(){
+        stepName = "";
+    }
+
+    Step(String stepName){
+        if(!stepName.contains(" "))
+            this.stepName = stepName;
+        else{
+            String[] str = stepName.split(" ", 0);
+            this.stepName = str[0];
+
+            boolean foundSecond = false;
+
+            for(int i = 1; i < str.length && !foundSecond; i++){
+                if(!str[i].equals(" ")) {
+                    foundSecond = true;
+                    this.defaultParameter = str[i];
+                }
+            }
+
+            if(!foundSecond)
+                throw new IllegalStateException("No parameter provided for step " + this.stepName);
+
+            if(this.defaultParameter.startsWith("'") || this.defaultParameter.startsWith("\""))
+                this.defaultParameter = this.defaultParameter.substring(1, this.defaultParameter.length() - 1);
```

Comment:
this has potential for a future bug. just because it starts with a '/" doesn't mean it ends with one. 
which also brings up the topic of handling multiline commands in the yaml...
- I don't think either of these need to be addressed in this PR, but maybe it should be documented in a jira ticket until you have a proper test case to address this

-----
also, Martin mentioned always using `{}`. Unless there's something in the Jenkins coding style guide that says otherwise, I agree with this point.

Reply:
The problem is the code tries to do lexical analysis and parsing of an input string. The problem starts on line 21 with the call to split. The split tries to tokenize the input by using a single space character as the separator, but if there is more than one whitespace of separation between tokens, it fails. You could try to split on `\\s+`, then check each element for empty, but you still have to deal with `sh "hello arg1 arg2"` and the cases where there are user input errors. Stronger lexing is needed here, or a different approach. I agree with Jeff to capture it in a Jira.

@gautamabhishek46 if you follow the [twitter coding style guidelines](https://github.com/twitter/commons/blob/master/src/java/com/twitter/common/styleguide.md), you will have a lot less negative comments on styling. Brace usage is also covered in the [1TBS](https://github.com/twitter/commons/blob/master/src/java/com/twitter/common/styleguide.md#indent-style) section.